hydra-main/src/test/java/com/addthis/hydra/job/web/JobRequestHandlerImplTest.java

```diff
@@ -123,6 +121,17 @@ public void createJob_ConfigTooLarge() throws Exception {
     }
 
     @Test
+    public void createJob_WithoutCreator() throws Exception {
+        Job job = new Job();
+        when(spawn.createJob("megatron", -1, Collections.<String> emptyList(), "default", "default-task", false)).thenReturn(job);
+
+        kv.add("config", "my job config");
+        kv.add("command", "default-task");
```

Comment:
You are testing the case where the front end passes empty string as creator. You should set up `kv` properly to simulate that:
```
kv.add("creator", "")
```
Accordingly, the stub for `Spawn.createJob` should be changed too because you are expecting an empty first argument which is creator:
```
when(spawn.createJob("", -1, Collections.<String> emptyList(), "default", "default-task", false)).thenReturn(job);
```